### ISSUE TITLE
Remove OSX dependency (aka support Linux)

### ISSUE THREAD

**Issue by trufae** [nowsecure/node-applesign#6]
As described in this project: https://github.com/saucelabs/isign

Looks like it is basically running openssl commands to do all that signing thing without depending on Apple hw or sw. (thanks @oleavr for the finding)

**Comment by trufae** [comment]
For a quick approach we can just write a bunch of shellscript or python lines to implement the `codesign` and `security` programs using `isign`. But ideally we should have those implemented in pure node.

Those are all the four commands used in applesign
```
codesign -v —no-strict binaryName
codesign —no-strict —fs identityFile —entitlements=$entitlementsfile binaryToSign
security find-identities -v -p codesigning
security cms -D -i embedded.mobileprovision
```

**Comment by dweinstein** [comment]
yeah I'm investing my time fixing some issues with isign so it could work out... though longer term I'd love to see something pure-node.

**Comment by trufae** [comment]
See 2d8caec33bee4cbdc58aff4b3a38501dd0693a59

**Comment by trufae** [comment]
Also see https://github.com/nowsecure/node-applesign/blob/master/tools.js

**Comment by mkayswork** [comment]
An pure node solution without os x dependency would be great, I've been looking for this a few weeks..

**Comment by trufae** [comment]
@mkayswork there's https://pypi.python.org/pypi/isign/ but from my experience it is pretty hard to setup and it is pretty buggy, so it is not a reliable option

**Comment by mkayswork** [comment]
Ty, I know that project because I have to use it to get rid of that OS X dependency. It's also working, but pure node would be far better for me.. 

The setup itself isn't that hard anymore, the issues also give good hints for some use cases.. ;)

**Comment by trufae** [comment]
You can help on this by writing separate standalone tools that mimic osx functionality but using isign under the hood to get applesign working outside osx.

After this it will be desirable to slowly reimplement those commands as node modules that can be used to replace the tools.js thing

**Comment by mkayswork** [comment]
I don't have enough knowledge regarding the signing process of os x. From what I've learned, the proccess is very complex. I managed to get it working on linux with isign https://pypi.python.org/pypi/isign/ . It's because It relies on OpenSSL which is only available as a third party solution on Windows. Since the code signing process is very complex, I don't want to rely on those, which is why I started using Isign.
